### ISSUE TITLE
Add a Gentoo bootstrap profile when a target machine exists

### ISSUE THREAD

**Issue by cuberhaus** [cuberhaus/dotfiles#20]
## Status
Future work. Do not implement or copy another distribution's package list until there is an actual Gentoo target machine.

## Goal
Add first-class Gentoo support that follows the repository's existing platform separation and Stow safety model.

## Prerequisites
Record these decisions from the target installation first:
- OpenRC or systemd.
- Gentoo profile, architecture, desktop/window-manager session, and display stack.
- Required USE flags, overlays, package keywords, and accepted binary/source package policy.
- User groups, services, hardware-specific packages, and machine-local exceptions.
- Which existing base packages and shell/editor configuration are genuinely portable.

## Future implementation scope
- Add a dedicated Gentoo bootstrap entry point and Gentoo-specific functions/package declarations.
- Share only platform-neutral helpers from `base_functions`.
- Integrate the safe fresh-checkout/Stow flow tracked in #13.
- Keep machine-local secrets and hardware exceptions out of the repository.
- Add profile detection support only after the explicit bootstrap profile works.
- Document required privileges, expected compilation time, recovery, and unsupported choices.

## Acceptance criteria
- The profile is derived from a documented real machine rather than assumptions.
- A clean Gentoo installation can complete the selected bootstrap without using Arch, Ubuntu, or macOS package commands.
- OpenRC/systemd behavior matches the chosen target and is not mixed implicitly.
- Repeated runs are idempotent where practical and report already-satisfied steps clearly.
- Stow conflicts are previewed and backed up according to #13.
- Tests cover profile selection, package-list parsing, unsupported configurations, and non-destructive dry paths.
- README and audit tooling list Gentoo only after the bootstrap is usable.

**Comment by cuberhaus** [comment]
> *This was generated by AI during triage.*

## Triage Notes

**What we have established so far:**

- This is a future enhancement for a Gentoo bootstrap profile.
- No Gentoo bootstrap entry or profile exists in the repository.
- The issue is intentionally deferred until a real Gentoo target exists.

**What we still need from you:**

- Target hardware, Gentoo profile, and architecture.
- OpenRC or systemd choice.
- Desktop or session stack, USE flags, overlays, and package policy.
- Required services and hardware-specific setup.

**Comment by cuberhaus** [comment]
> *This was generated by AI during triage.*

## Triage Notes

**What we've established so far:**

- No Gentoo target machine or active bootstrap target exists yet.
- A generic profile should not be invented without a real target and reproducible requirements.
- The request remains a future enhancement rather than an implementation task.

**What we still need from you (@reporter):**

- A target machine or concrete installation workflow.
- The Gentoo profile, init system, desktop/session, USE flags, and services that must be supported.
